File: quiz_app_flutter/lib/domain/mapper/quiz_mapper.dart
import '../../data/model/quiz/quiz_list_dto.dart' as QuizListDto;
import '../entities/quiz/quiz_data.dart';

extension QuizItemDtoX on QuizListDto.QuizItemDto {
  QuizData toDomain() => QuizData(
    questionId: questionId,
    question: question,
    answerCellType: answerCellType,
    selectedOptions: selectedOptions,
    answerSectionTitle: answerSectionTitle,
    explanation: explanation,
    answerCellList: answerCellList.map((e) => e.toDomain()).toList(),
    correctAnswer: correctAnswer.toDomain(),
  );
}

extension AnswerCellDtoX on QuizListDto.AnswerCellDto {
  QuizAnswerCell toDomain() => QuizAnswerCell(
    answerId: answerId,
    questionId: questionId,
    data: data,
    isItAnswer: isItAnswer,
    position: position,
  );
}

extension CorrectAnswerDtoX on QuizListDto.CorrectAnswerDto {
  QuizCorrectAnswer toDomain() => QuizCorrectAnswer(
    questionId: questionId,
    answerId: answerId,
    answer: answer,
    explanation: explanation,
  );
}

File: quiz_app_kmp/composeApp/src/commonMain/kotlin/com/tnm/quizappkmp/data/mapper/QuizMapper.kt
package com.tnm.quizappkmp.data.mapper

import com.tnm.quizappkmp.data.dto.QuizListDto
import com.tnm.quizappkmp.domain.model.quiz.QuizData


fun QuizListDto.QuizItemDto.toDomain() = QuizData(
    questionId = questionId,
    question = question,
    answerCellType = answerCellType,
    selectedOptions = selectedOptions,
    answerSectionTitle = answerSectionTitle,
    explanation = explanation,
    answerCellList = answerCellList.map { it.toDomain() },
    correctAnswer = correctAnswer.toDomain()
)

fun QuizListDto.AnswerCellDto.toDomain() = QuizData.AnswerCell(
    answerId = answerId,
    questionId = questionId,
    data = data,
    isItAnswer = isItAnswer,
    position = position
)

fun QuizListDto.CorrectAnswerDto.toDomain() = QuizData.CorrectAnswer(
    questionId = questionId,
    answerId = answerId,
    answer = answer,
    explanation = explanation
)
